src/measure.jl

```diff
@@ -1,46 +1,46 @@
-using StatsBase, StaticArrays, BitBasis
+using StatsBase, StaticArrays, BitBasis, Random
 export measure,
     measure!,
     measure_remove!,
     measure_collapseto!,
     select,
     select!
 
-_measure(pl::AbstractVector, nshots::Int) = sample(0:length(pl)-1, Weights(pl), nshots)
-function _measure(pl::AbstractMatrix, nshots::Int)
+_measure(rng::AbstractRNG, pl::AbstractVector, nshots::Int) = sample(rng, 0:length(pl)-1, Weights(pl), nshots)
+function _measure(rng::AbstractRNG, pl::AbstractMatrix, nshots::Int)
     B = size(pl, 2)
     res = Matrix{Int}(undef, nshots, B)
     for ib=1:B
-        @inbounds res[:,ib] = _measure(view(pl,:,ib), nshots)
+        @inbounds res[:,ib] = _measure(rng, view(pl,:,ib), nshots)
     end
     return res
 end
 
-YaoBase.measure(::ComputationalBasis, reg::ArrayReg{1}, ::AllLocs; nshots::Int=1) = _measure(reg |> probs, nshots)
+YaoBase.measure(rng::AbstractRNG, ::ComputationalBasis, reg::ArrayReg{1}, ::AllLocs; nshots::Int=1) = _measure(rng, reg |> probs, nshots)
```

Comment:
I think you need to put a default RNG (MT RNG) there, or it breaks with previous version.

Reply:
CI should just pass, if you add a default RNG.

Reply:
Tests passed locally. So what is the problem.

Reply:
Oh, OK. I think I need to merge YaoBase first.